### ISSUE TITLE
api-readonly design questions

### ISSUE THREAD

**Issue by Habbie** [PowerDNS/pdns#6786]
- Program: Authoritative, Recursor
 - Issue type: Design discussion

### Short description
#6781 points out that the `rectify-zone` API endpoint works even when `api-readonly` is set.  This is not the first time we 'missed a spot'.

Two proposals have been made informally:
1. when the request is not GET (i.e. it is POST/PUT/PATCH), and `api-readonly` is set, never even dispatch the call to the defined handlers
2. drop `api-readonly`

In the interest of reducing complexity, I would currently prefer option 2.

Thoughts?

**Comment by pieterlexis** [comment]
> In the interest of reducing complexity, I would currently prefer option 2.

Agree. The API should only do authentication (based on the `X-API-Key` header). All authorization (and fine-grained authentication) should be done with a front-end proxy program. As every user has different demands.
